.github/workflows/nox.yml

```diff
@@ -0,0 +1,50 @@
+name: Nox
+
+on: [pull_request]
```

Comment:
I would additionally run it on pushes (at least) to master. 

Can be done as follows:

```
on: 
  push:
    branches:
      - "master"
  pull_request:
```

Reply:
Added (and also workflow_dispatch, so we can run it manually if we want).
I'm wondering a bit if running on master is really needed, though.  Instead I'd rather set branch rules such that one can only merge via pull request if the branch is up-to-date and checks have passed.  In this case running again on master after the merge would just repeat the same checks.

Reply:
That sounds acceptable if merges to master really ever only happen that way. Also, does the Github UI render the CI checkmark correctly on the main page if it's only run on a branch?

Reply:
I think it doesn't show any check mark on the main page.  If we want this, we would probably need to add a badge to the README.  In that case, it is probably really necessary to run on push to master.